Barotrauma/BarotraumaShared/SharedSource/StatusEffects/StatusEffect.cs

```diff
@@ -2281,15 +2285,74 @@ void OnItemSpawned(Item newItem, ItemSpawnInfo itemSpawnInfo)
+                    }
+
+                    if (containerPrefab == null)
+                    {
+                        if (saveItemsContainerIdentifier == null)
+                        {
+                            DebugConsole.NewMessage("Could not spawn a container for a removed character's items. No item with the tag \"despawncontainer\" or the identifier \"metalcrate\" found.", Color.Red);
+                        }
+                        else
+                        {
+                            DebugConsole.NewMessage($"Could not spawn a container for a removed character's items. No item with the identifier “{saveItemsContainerIdentifier}” found.", Color.Red);
+                        }
+                    }
+                    else
+                    {
+                        Entity.Spawner?.AddItemToSpawnQueue(containerPrefab, position, onSpawned: onItemContainerSpawned);
+                    }
+
+                    void onItemContainerSpawned(Item item)
+                    {
+                        if (character.Inventory == null) { return; }
+
+                        item.UpdateTransform();
+                        item.AddTag("name:" + character.Name);
+                        if (character.Info?.Job != null) { item.AddTag($"job:{character.Info.Job.Name}"); }
+
+                        var itemContainer = item?.GetComponent<ItemContainer>();
+                        if (itemContainer == null) { return; }
+                        List<Item> inventoryItems = new List<Item>(character.Inventory.AllItemsMod);
+                        foreach (Item inventoryItem in inventoryItems)
+                        {
+                            if (!itemContainer.Inventory.TryPutItem(inventoryItem, user: null, createNetworkEvent: true))
+                            {
+                                //if the item couldn't be put inside the despawn container, just drop it
+                                inventoryItem.Drop(dropper: character, createNetworkEvent: true);
+                            }
+                        }
+                    }
+                }
                 for (int i = 0; i < targets.Count; i++)
                 {
                     var target = targets[i];
                     if (target is Character character)
                     {
+                        if (removeCharacterAndSaveItems) // Before removing the character, place their items in a container.
+                        {
+                            SpawnDespawnContainer(character);
+                        }
                         Entity.Spawner?.AddEntityToRemoveQueue(character);
                     }
                     else if (target is Limb limb)
                     {
+                        if (removeCharacterAndSaveItems)
+                        {
+                            SpawnDespawnContainer(limb.character);
+                        }
```

Comment:
There's some code duplication here, could be simplified by storing the target character in a variable instead of implementing the logic separately for characters and limbs.